### ISSUE TITLE
How to change caret position after changing text alignment?

### ISSUE THREAD

**Issue by The28AWG** [superlistapp/super_editor#1407]
![image](https://github.com/superlistapp/super_editor/assets/1275371/5213086f-7366-4f03-a493-4308f2e870f5)
the screenshot shows the alignment result TextAlign.center -> TextAlign.right
It may be possible to force the caret position to be updated?

**Comment by The28AWG** [comment]
I am using super_editor version 0.2.6
It looks like the algorithm for working with the caret has been slightly changed in the repository

**Comment by The28AWG** [comment]
![image](https://github.com/superlistapp/super_editor/assets/1275371/6572f101-0d84-45d7-be1f-51d07cdeb51c)
I couldn't run the main branch, there are too many errors that I can’t fix, so the following code concerns the stable branch.
https://github.com/superlistapp/super_editor/blob/d2f50c09d276f607e90e241a6d40dfde3fc0249b/super_editor/example/lib/demos/example_editor/example_editor.dart#L410
I had to add the following parameters to SuperEditor because without them the toolbar loses focus:
```dart
            selectionPolicies: SuperEditorSelectionPolicies(
              clearSelectionWhenEditorLosesFocus: false,
              clearSelectionWhenImeConnectionCloses: false,
            ),
```
https://github.com/superlistapp/super_editor/blob/d2f50c09d276f607e90e241a6d40dfde3fc0249b/super_editor/example/lib/demos/example_editor/_toolbar.dart#L436
add
```dart
    widget.closeToolbar();
```
This small modification is enough to ensure that the toolbar works on desktop and the focus does not disappear when hovering over the text alignment.
This allows you to test the text alignment behavior. We see that caret position is incorrect.

**Comment by matthew-carroll** [comment]
I think you still haven't described the actual behavior that's causing the problem. Please provide reproduction steps to see the issue in your screenshot.

**Comment by The28AWG** [comment]
[text align.webm](https://github.com/superlistapp/super_editor/assets/1275371/2c06c852-29d9-44f0-953f-f55e61ec7dc7)

The recording shows that the caret did not change its position after the text was aligned.

**Comment by rutvik110** [comment]
Fixed by #1714
